### PR TITLE
[EXPERIMENT] fix(cpu-profiler): use per-CPU BPF maps

### LINKED ISSUE #520: bpf: Make sure there is no concurrent access issue with CPU profiler
By migrating to Rust, we might introduce a regression for atomic counters.
The easier way to fix the issue is to use pre-CPU BPF maps.

xref: https://github.com/aquasecurity/libbpfgo/pull/156/files

### PR DESCRIPTION
**Pull request by maxbrunet**
Not sure how to test it effectively tho

Closes #520

### CODE REVIEW

**Comment by kakkoyun** [comment]
cc @javierhonduco

**Review by kakkoyun — APPROVED** [review]
LGTM.

As we discussed in the issue, we will benchmark against other approaches and pick one of them for the releases.

Thanks for the amazing work!

File: pkg/profiler/profiler.go [hunk 1]
```diff
@@ -748,11 +748,17 @@ func (p *CgroupProfiler) readKernelStack(kernelStackID int32, stack *stack) erro
 
 // readValue reads the value of the given key from the counts ebpf map.
 func (p *CgroupProfiler) readValue(keyBytes []byte) (uint64, error) {
-	valueBytes, err := p.bpfMaps.counts.GetValue(unsafe.Pointer(&keyBytes[0]))
-	if err != nil {
+	valueBytes := make([]byte, 8*runtime.NumCPU()) // u64 size (8 bytes) x NumCPU
```
Comment (brancz) [inline pkg/profiler/profiler.go:751]: This looks fairly expensive to do for every single value, especially if we're now multiplying it by the number of CPUs. I'm happy to merge as is and base optimizations on profiling data though.

**Comment by maxbrunet** [comment]
> @maxbrunet This is somewhat deprecated now. Should we close it or do you want to update it?

We can close, it'll be probably easier to create a new one if we are still interested in the experiment

**Pull request closed without merging**